### PR TITLE
Problem solved and repo rebased

### PR DESCRIPTION
**Pull request by D0048**
Changes since last pull request:
1. Added random salt generation for passwords.
2. Changed unit of delay on blinking screen from u second to m second.  
3. Rebased and summarized all the changes applied till now. 

Thanks for checking and pointing out the errors. : )

### CODE REVIEW

**Comment by D0048** [comment]
Add:
I changed up the handling of arguments so there will be no more confusion.

File: xtrlock.c [hunk 1]
```diff
@@ -18,196 +18,373 @@
...
+int passwordok(const char* s)
+{
+        /* simpler, and should work with crypt() algorithms using longer
+           salt strings (like the md5-based one on freebsd).  --marekm */
+        debug_print("%s, %i\n", s, (int)strlen(s));
+
+        if (cust_pw_setting.enable) {
+                debug_print("Entered_de: %s\n", s);
+                debug_print("Original_de: %s\n", cust_pw_setting.pwd);
+                char* enter = strdup(crypt(s, cust_pw_setting.pwd));
```
Comment (jaseg) [inline xtrlock.c]: BTW, technically I don't think you have to strdup the crypt result if you're only feeding it straight into strcmp anyway.

File: xtrlock.c [hunk 2]
```diff
@@ -18,196 +18,373 @@
...
+                if (NULL == enter) {
+                        fprintf(stderr, "\"strdup\" or \"crypt\":%s\n", strerror(errno));
+                        return false;
+                }
+                debug_print("Entered:  %s\n", enter);
+                debug_print("Original: %s\n", original);
+                unsigned int i = strcmp(enter, original);
+                free(enter);
+                return !i;
+        } else { /*salt in the second argument seems to be automatically used by crypt*/
```
Comment (jaseg) [inline xtrlock.c]: This code is pretty redundant, you're doing crypt/strcmp in two places here. You could simplify this.

**Review by jaseg — APPROVED** [review]
Looks good to me.

File: xtrlock.c (hunk 2 above)
Comment (D0048) [reply xtrlock.c]: I've removed the redundant code. If there are no more problems, I will be drawing a release and be started on the next version. (I read the article [here](https://www.jwz.org/xscreensaver/toolkits.html) and got really impressed.  Maybe I will try adding a daemon process to this program in the next version.)

**Comment by D0048** [comment]
Now I've started working on the second version. Should this merge be executed?

**Pull request closed without merging**